### PR TITLE
Improvements for the OpenAPI Generator Server Guide

### PR DESCRIPTION
**Pull request by andriy-dmytruk**
* Added information about what is OpenAPI and why use it.
* Added more description to many sections. Renamed some sections to be more descriptive of what the user will get after finishing it.
* Modified some of the common snippets. In particular, added the necessary mysql dependencies to some of them.

Some detailed comments:
* The `common-data-jdbc-mysql-configuration.adoc` is similar to `common-data-jdbc-mysql.adoc` but I couldn't change the original, since other guides use it. I added changes to the `application.yml` that are required for the application to run correctly. Also, added how to start MySQL Docker server, which seems very useful for anyone completely new to server development.
* I modified the `common-flyway.adoc` to add mysql dependency if required. The easiest way was to set a variable: `:database: mysql` that would trigger adding of the dependency. But I can add support for snippet parameters if it is better.
* The complete solution doesn't seem entirely logical. There is a bulletin list with just one item and the complete solution is in bold, but it is not a link or a variable. Can we change it?

### CODE REVIEW

File: guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc [hunk 1]
```diff
@@ -2,17 +2,39 @@ common:header-top.adoc[]
...
+
+Then we will add internal logic to the API and test our implementation.
+
+=== What OpenAPI Is
+
+The link:https://oai.github.io/Documentation/start-here.html[OpenAPI Specification] defines a format for uniquely describing REST APIs, that is both human- and machine-readable. Later in this guide we will discover the structure of documents in the OpenAPI format. We will also create such document for our desired API. Note, that we will refer to the document describing our API as **API definition** file.
+
+=== Advantages Of OpenAPI
+
+* It provides a unique way of describing a REST API that is easy to understand and modify. It is the most broadly adopted industry standard for describing new APIs and has the most developed tooling ecosystem.
```
Comment (sdelamo) [inline guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc:15]: Is this original copy or is it a quote?

**Comment by sdelamo** [comment]
> * I modified the `common-flyway.adoc` to add mysql dependency if required. The easiest way was to set a variable: `:database: mysql` that would trigger adding of the dependency. But I can add support for snippet parameters if it is better.


I was not aware of this. 

```adoc
:database: mysql
common:flyway.adoc[]
```

and inside flyway he has done: 

```adoc
Add the following snippet to include the necessary dependencies:

:dependencies:
dependency:micronaut-flyway[groupId=io.micronaut.flyway]
ifeval::["{database}" == "mysql"]
dependency:flyway-mysql[groupId=org.flywaydb,scope=runtimeOnly]
endif::[]
```

what do you think @burtbeckwith ?

**Comment by sdelamo** [comment]
> * The `common-data-jdbc-mysql-configuration.adoc` is similar to `common-data-jdbc-mysql.adoc` but I couldn't change the original, since other guides use it. I added changes to the `application.yml` that are required for the application to run correctly. Also, added how to start MySQL Docker server, which seems very useful for anyone completely new to server development.

Can we have a single snippet that works for every guide?

**Review by sdelamo — APPROVED** [review]

File: guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc (hunk 1 above)
Comment (andriy-dmytruk) [reply guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc:15]: The information is from OpenAPI website, but sentence is rephrased. For example they have it here: https://oai.github.io/Documentation/introduction.html

> The following section lists the benefits of OpenAPI, the most widely used specification when creating new APIs.

**Pull request merged**